packages/python/plotly/_plotly_utils/basevalidators.py

```diff
@@ -8,6 +8,7 @@
 import re
 import sys
 import warnings
+import narwhals.stable.v1 as nw
```

Comment:
Concerning stability and testing purposes, wondering if it's better to load specific version instead?

Reply:
Would you suggest to pin a specific narwhals version in the requirements?

We try to make [some promises](https://narwhals-dev.github.io/narwhals/backcompat/) of non-breaking changes with the stable API. Additionally, we test each change against the main/master branches of the downstream libraries that adopted narwhals in [github action](https://github.com/narwhals-dev/narwhals/blob/main/.github/workflows/downstream_tests.yml). 

Of course this is not perfect, but gives us some confidence when we make a change and a release

Reply:
hi @archmoj ! nice to meet you, and thanks for taking a look 🙏 

I'd advise against pinning the Narwhals version exactly, and that's because it might conflict with other tools which are using Narwhals

For example, Altair currently has `narwhals>=1.5.2`, and this PR has `narwhals>=1.12.0`. So, a user wanting to install both just needs to have `narwhals>=1.12.0`, and there's no issues

If Plotly were to pin Narwhals exactly (say, `narwhals==1.12.0`), and Altair were to do the same (say, `narwhals==1.10.0`), then users wouldn't be able to install both Altair and Plotly together, which would be a pity

Indeed, as @FBruzzesi says, the intention behind `narwhals.stable.v1` is to have a stable and perfectly backwards-compatible API, in order to give downstream libraries (e.g. Plotly, Altair, Marimo, ...) the confidence to not pin the Narwhals version exactly, allowing for multiple major libraries to have Narwhals as a dependency without forbidding users to install them all together

Furthermore, for major libraries, we run downstream tests in the Narwhals CI to check that any change we add won't break things for you

Happy to discuss further if you like of course (and to take a call if you'd like to discuss this or other topics) 🤗

Reply:
Would the following comment under the import be enough?

```
import narwhals.stable.v1 as nw
# The reason to use narwhals.stable.v1 is to have a stable and perfectly 
# backwards-compatible API, hence the confidence to not pin the Narwhals version exactly,
# allowing for multiple major libraries to have Narwhals as a dependency without
# forbidding users to install them all together due to dependency conflicts.
```

It's a TL;DR of what Marco mentioned